util.go

```diff
@@ -155,9 +155,14 @@ func unmarshallConfigReader(in io.Reader, c map[string]interface{}, configType s
 		}
 
 	case "toml":
-		if _, err := toml.Decode(buf.String(), &c); err != nil {
+		tomlTree, err := toml.Load(buf.String())
+		if err != nil {
 			return ConfigParseError{err}
 		}
+		tomlMap := tomlTree.ToMap()
```

Comment:
It creates an extra map. If this is a critical path, I can try to send a patch to `pelletier/go-toml` to avoid the extra map.